test/e2e/dnslookup_test.go

```diff
@@ -130,20 +133,32 @@ func TestMain(m *testing.M) {
 	err = writeDNSHammerFile(filepath.Join(tmpDir, dnsHammerCNAMERecords), cnameRecords)
 	requireNoErrorf(err, "Failed generating CNAME test data")
 
-	logrus.Infof("Dowloading %v wsl distro tarball", wslTarballName)
-	tarballPath := filepath.Join(tmpDir, wslTarballName)
-
-	err = downloadFile(tarballPath, wslTarballURL)
-	requireNoErrorf(err, "Failed to download wsl distro tarball: %v", err)
+	tarballPath := filepath.Join(os.TempDir(), wslTarballName)
+	s, err := os.Stat(tarballPath)
+	// Only download the tarball if
+	// 1) it doesn't exist
+	// 2) the cache is expired > 60 min
+	if errors.Is(err, os.ErrNotExist) {
+		err = downloadFile(tarballPath, wslTarballURL)
+		requireNoErrorf(err, "Failed to download wsl distro tarball %v", wslTarballName)
+	} else {
+		if s.ModTime().Before(time.Now().Add(time.Hour * -1)) {
```

Comment:
If `err` is non-`nil` (and not `ErrNotExist`) but we haven't timed out yet, this drops the error on the floor.